packages/fuel-gauge/src/advanced-logging.test.ts

```diff
@@ -154,23 +179,38 @@ describe('Advanced Logging', () => {
       'fuelfuel',
     ];
 
-    beforeAll(async () => {
-      wallet = await generateTestWallet(provider, [[500_000, baseAssetId]]);
-    });
-
     it('when using InvacationScope', async () => {
-      const callTest = await setupCallTest({ cache: false });
-      const configurable = await setupConfigurable({ cache: false });
-      const coverage = await setupCoverage({ cache: false });
+      using launched = await launchTestNode({
+        contractsConfigs: [
+          { deployer: AdvancedLoggingAbi__factory, bytecode: AdvancedLoggingAbiHex },
+          {
+            deployer: AdvancedLoggingOtherContractAbi__factory,
+            bytecode: AdvancedLoggingOtherContractAbiHex,
+          },
+          { deployer: CallTestContractAbi__factory, bytecode: CallTestContractAbiHex },
+          { deployer: ConfigurableContractAbi__factory, bytecode: ConfigurableContractAbiHex },
+          { deployer: CoverageContractAbi__factory, bytecode: CoverageContractAbiHex },
+        ],
+      });
+
+      const {
+        contracts: [
+          advancedLogContract,
+          otherAdvancedLogContract,
+          callTest,
+          configurable,
+          coverage,
+        ],
+      } = launched;
 
       const { waitForResult } = await callTest
         .multiCall([
-          advancedLogContract.functions
-            .test_log_from_other_contract(10, otherLogId)
+          (advancedLogContract as AdvancedLoggingAbi).functions
+            .test_log_from_other_contract(10, otherAdvancedLogContract.id.toB256())
             .addContracts([otherAdvancedLogContract]),
-          callTest.functions.boo(testStruct),
-          configurable.functions.echo_struct(),
-          coverage.functions.echo_str_8('fuelfuel'),
+          (callTest as CallTestContractAbi).functions.boo(testStruct),
+          (configurable as ConfigurableContractAbi).functions.echo_struct(),
+          (coverage as CoverageContractAbi).functions.echo_str_8('fuelfuel'),
```

Comment:
What's up with all these `contract as *Abi` castings? 👀 

Is this a missing detail on the inner workings and type-gymnastics of the launcher utility?

Reply:
Seems like I missed this because I didn't test it out with _different_ contract factories in the `contractConfigs` array but put two _same_ factories and thought that the type inference works correctly while it was just a discriminated union of two same types which resulted in one type and me thinking it works. I'm on it in #2754.